src/plugins/intel_gna/src/frontend/scale_factor_calc.cpp

```diff
@@ -261,7 +261,7 @@ bool ScaleFactorCalculator::requantizeInput(InferenceEngine::CNNLayerPtr input,
     if (info.isWeightableIdentity() && !AreFpEq(quantDataForInputLayer->_weights_quant.GetScale(), 1.0f)) {
         auto reducer = std::max(1.0f, quantDataForInputLayer->_dst_quant.GetScale() / newOutputScale);
         auto newWeightsScale = std::max(1.0f, quantDataForInputLayer->_weights_quant.GetScale() / reducer);
-        quantDataForInputLayer->_weights_quant.SetScale(static_cast<int32_t>(newWeightsScale));
+        quantDataForInputLayer->_weights_quant.SetScale(trunc(newWeightsScale));
```

Comment:
don't you have warning here?
It looks like newWeightsScale is double and trunc returns double, and later you pass it to SetScale which takes float.

Reply:
newWeightsScale is float, so trunc also returns a float.